File: src/simulation/FitnessResolver.java
package simulation;

import neat.Genotype;
import worldbuilding.BodySettings;

import java.util.ArrayList;

/**
 * Created by colander on 2/1/17.
 * DEPRECATED
 * Class used to compute all fitnesses of genotypes in and ArrayList.
 */
public class FitnessResolver {

    ArrayList<Genotype> genotypes;
    BodySettings settings;

    FitnessResolver(ArrayList<Genotype> genotypes, BodySettings settings) {
        this.genotypes = genotypes;
        this.settings = settings;
    }

    @Deprecated
    public ArrayList<FitnessResult> resolve() {
        ArrayList<FitnessResult> results = new ArrayList<>();
        for (int i = 0; i < genotypes.size(); i++) {
            FitnessTest test = new FitnessTest(genotypes.get(i), settings, i);
            results.add(new FitnessResult(test.compute().result, genotypes.get(i)));
        }
        return results;
    }
}

File: src/main/java/com/janboucek/crawler/simulation/FitnessResolver.kt
package com.janboucek.crawler.simulation

import com.janboucek.crawler.neat.Genotype
import com.janboucek.crawler.worldbuilding.BodySettings
import java.util.*

/**
 * Created by colander on 2/1/17.
 * DEPRECATED
 * Class used to compute all fitnesses of genotypes in and ArrayList.
 */
open class FitnessResolver internal constructor(var genotypes: ArrayList<Genotype>, var settings: BodySettings) {
    open fun resolve(): List<FitnessResult>? {
        val results = ArrayList<FitnessResult>()
        for (i in genotypes.indices) {
            val test = FitnessTest(genotypes[i], settings, i)
            results.add(FitnessResult(test.compute().result, genotypes[i]))
        }
        return results
    }

}